src/utilities/filetypes/ForwardTranslatorOptions.cpp

```diff
@@ -0,0 +1,463 @@
+    return m_is_ip_tabular_output_defaulted;
+  }
+
+  void ForwardTranslatorOptions_Impl::setIPTabularOutput(bool iPTabularOutput) {
+    m_ip_tabular_output = iPTabularOutput;
+    m_is_ip_tabular_output_defaulted = false;
+  }
+
+  void ForwardTranslatorOptions_Impl::resetIPTabularOutput() {
+    m_ip_tabular_output = DEFAULT_IP_TABULAR_OUTPUT;
+    m_is_ip_tabular_output_defaulted = true;
+  }
+
+  bool ForwardTranslatorOptions_Impl::excludeLCCObjects() const {
+    return m_no_lifecyclecosts;
+  }
+
+  bool ForwardTranslatorOptions_Impl::isExcludeLCCObjectsDefaulted() const {
+    return m_is_no_lifecyclecosts_defaulted;
+  }
+
+  void ForwardTranslatorOptions_Impl::setExcludeLCCObjects(bool excludeLCCObjects) {
+    m_no_lifecyclecosts = excludeLCCObjects;
+    m_is_no_lifecyclecosts_defaulted = false;
+  }
+
+  void ForwardTranslatorOptions_Impl::resetExcludeLCCObjects() {
+    m_no_lifecyclecosts = DEFAULT_NO_LIFECYCLECOSTS;
+    m_is_no_lifecyclecosts_defaulted = true;
+  }
+
+  bool ForwardTranslatorOptions_Impl::excludeSQliteOutputReport() const {
+    return m_no_sqlite_output;
+  }
+
+  bool ForwardTranslatorOptions_Impl::isExcludeSQliteOutputReportDefaulted() const {
+    return m_is_no_sqlite_output_defaulted;
+  }
+
+  void ForwardTranslatorOptions_Impl::setExcludeSQliteOutputReport(bool excludeSQliteOutputReport) {
+    m_no_sqlite_output = excludeSQliteOutputReport;
+    m_is_no_sqlite_output_defaulted = false;
+  }
+
+  void ForwardTranslatorOptions_Impl::resetExcludeSQliteOutputReport() {
+    m_no_sqlite_output = DEFAULT_NO_SQLITE_OUTPUT;
+    m_is_no_sqlite_output_defaulted = true;
+  }
+
+  bool ForwardTranslatorOptions_Impl::excludeHTMLOutputReport() const {
+    return m_no_html_output;
+  }
+
+  bool ForwardTranslatorOptions_Impl::isExcludeHTMLOutputReportDefaulted() const {
+    return m_is_no_html_output_defaulted;
+  }
+
+  void ForwardTranslatorOptions_Impl::setExcludeHTMLOutputReport(bool excludeHTMLOutputReport) {
+    m_no_html_output = excludeHTMLOutputReport;
+    m_is_no_html_output_defaulted = false;
+  }
+
+  void ForwardTranslatorOptions_Impl::resetExcludeHTMLOutputReport() {
+    m_no_html_output = DEFAULT_NO_HTML_OUTPUT;
+    m_is_no_html_output_defaulted = true;
+  }
+
+  bool ForwardTranslatorOptions_Impl::excludeVariableDictionary() const {
+    return m_no_variable_dictionary;
+  }
+
+  bool ForwardTranslatorOptions_Impl::isExcludeVariableDictionaryDefaulted() const {
+    return m_is_no_variable_dictionary_defaulted;
+  }
+
+  void ForwardTranslatorOptions_Impl::setExcludeVariableDictionary(bool excludeVariableDictionary) {
+    m_no_variable_dictionary = excludeVariableDictionary;
+    m_is_no_variable_dictionary_defaulted = false;
+  }
+
+  void ForwardTranslatorOptions_Impl::resetExcludeVariableDictionary() {
+    m_no_variable_dictionary = DEFAULT_NO_VARIABLE_DICTIONARY;
+    m_is_no_variable_dictionary_defaulted = true;
+  }
```

Comment:
getter, setters, isXXDefaulted and resetXXX are provided, similar to model objects. Except it's not IDD based but uses explicit members